src/graph/resolver.rs

```diff
@@ -242,81 +251,101 @@ impl Resolver {
+                    .or(a_exec.depth_stencil_load)
+                    .or(a_exec.depth_stencil_store)
+                    .or_else(|| {
+                        a_exec
+                            .depth_stencil_resolve
+                            .map(|(attachment, ..)| attachment)
                     })
+                    .or_else(|| a_exec.depth_stencil_clear.map(|(attachment, _)| attachment));
+
+                let b_depth_stencil = b_exec
+                    .depth_stencil_attachment
+                    .or(b_exec.depth_stencil_load)
+                    .or(b_exec.depth_stencil_store)
                     .or_else(|| {
-                        rhs_exec
-                            .color_resolves
-                            .get(attachment_idx)
-                            .map(|(attachment, _)| attachment)
-                    });
+                        b_exec
+                            .depth_stencil_resolve
+                            .map(|(attachment, ..)| attachment)
+                    })
+                    .or_else(|| b_exec.depth_stencil_clear.map(|(attachment, _)| attachment));
 
-                if !Attachment::are_compatible(Some(*lhs_attachment), rhs_attachment.copied()) {
-                    trace!("  incompatible color attachments");
+                if !Attachment::are_compatible(a_depth_stencil, b_depth_stencil) {
+                    trace!("  incompatible depth/stencil attachments");
 
                     return false;
-                } else {
-                    common_color_attachment = true;
                 }
-            }
-
-            // Compare depth/stencil attachments for compatibility
-            let lhs_depth_stencil = lhs_exec
-                .depth_stencil_attachment
-                .or(lhs_exec.depth_stencil_load)
-                .or(lhs_exec.depth_stencil_store)
-                .or_else(|| {
-                    lhs_exec
-                        .depth_stencil_resolve
-                        .map(|(attachment, ..)| attachment)
-                })
-                .or_else(|| {
-                    lhs_exec
-                        .depth_stencil_clear
-                        .map(|(attachment, _)| attachment)
-                });
-            if !Attachment::are_compatible(lhs_depth_stencil, rhs_depth_stencil) {
-                trace!("  incompatible depth/stencil attachments");
 
-                return false;
+                if a_depth_stencil.is_some() && b_depth_stencil.is_some() {
+                    common_depth_attachment = true;
+                } else {
+                    return false;
+                }
             }
 
-            let common_depth_attachment =
-                lhs_depth_stencil.is_some() && rhs_depth_stencil.is_some();
-
             // Keep color and depth on tile.
-            if common_color_attachment || common_depth_attachment {
+            if common_color_attachment && common_depth_attachment {
```

Comment:
I wonder if this should more pessimistically do 
```rs
if !common_color_attachment || !common_depth_attachment {
    return false;
}
```
Then things below can also return false. I guess those things don't exist yet. But it could be confusing if they are added.